`#warning` in struct causes compiler debug message

**Describe the bug**
The compiler emits what looks like a compiler debug message when encountering `#warning` in a struct.
It also duplicates the `#warning` message in output. 

**To Reproduce**
Steps to reproduce the behavior:
1. put the following in `main.swift` of a SPM executable
2. `swift run`

```swift
struct S {
  #warning("a warning here")
}
```

**Actual behavior**
```out
Building for debugging...
/private/tmp/user/Sources/user/main.swift:2:12: warning: a warning here
  #warning("a warning here")
           ^~~~~~~~~~~~~~~~
/private/tmp/user/Sources/user/main.swift:2:12: warning: a warning here
  #warning("a warning here")
           ^~~~~~~~~~~~~~~~
Unhandled decl:
(pound_diagnostic_decl range=[/private/tmp/user/Sources/user/main.swift:2:3 - line:2:28] kind=warning
    (string_literal_expr type='<null>' encoding=utf8 value="a warning here" builtin_initializer=**NULL** initializer=**NULL**))
[3/3] Linking user
Build complete! (0.95s)
```

**Expected behavior**
```out
Building for debugging...
/private/tmp/user/Sources/user/main.swift:2:12: warning: a warning here
  #warning("a warning here")
           ^~~~~~~~~~~~~~~~
[3/3] Linking user
Build complete! (0.95s)
```

**Environment (please complete the following information):**
`swift --version`
swift-driver version: 1.45.2 Apple Swift version 5.6 (swiftlang-5.6.0.323.62 clang-1316.0.20.8)
Target: x86_64-apple-macosx12.0

--- a/lib/APIDigester/ModuleAnalyzerNodes.cpp
+++ b/lib/APIDigester/ModuleAnalyzerNodes.cpp
@@ -1834,6 +1834,8 @@ SwiftDeclCollector::addMembersToRoot(SDKNode *Root, IterableDeclContext *Context
       // All containing variables should have been handled.
     } else if (isa<IfConfigDecl>(Member)) {
       // All containing members should have been handled.
+    } else if (isa<PoundDiagnosticDecl>(Member)) {
+      // All containing members should have been handled.
     } else if (isa<DestructorDecl>(Member)) {
       // deinit has no impact.
     } else if (isa<MissingMemberDecl>(Member)) {

